rust/lance-index/src/vector/hnsw.rs

```diff
@@ -289,13 +298,27 @@ impl HNSW {
         let num_layers = self.levels.len();
         for level in self.levels.iter().rev().take(num_layers - 1) {
             let candidates = beam_search(level, &ep, query, 1)?;
-            ep = select_neighbors(&candidates, 1).map(|(_, id)| id).collect();
+            ep = if self.use_select_heuristic {
+                select_neighbors_heuristic(level, query, &candidates, 1, false, true)
```

Comment:
should we just use this as default? Or just remove the simple algo entirely?

Reply:
heuristic search is default

https://github.com/lancedb/lance/pull/1991/files#diff-c6be5e3d56aeb264a665c87d8f7b7af9ff2177982aade02bf28c4515d1974eeaR264